sha1/src/compress/loongarch64_asm.rs

```diff
@@ -0,0 +1,262 @@
+            round2!("$t2", "$t3", "$t4", "$t0", "$t1", 48),
+            round2!("$t1", "$t2", "$t3", "$t4", "$t0", 49),
+            round2!("$t0", "$t1", "$t2", "$t3", "$t4", 50),
+            round2!("$t4", "$t0", "$t1", "$t2", "$t3", 51),
+            round2!("$t3", "$t4", "$t0", "$t1", "$t2", 52),
+            round2!("$t2", "$t3", "$t4", "$t0", "$t1", 53),
+            round2!("$t1", "$t2", "$t3", "$t4", "$t0", 54),
+            round2!("$t0", "$t1", "$t2", "$t3", "$t4", 55),
+            round2!("$t4", "$t0", "$t1", "$t2", "$t3", 56),
+            round2!("$t3", "$t4", "$t0", "$t1", "$t2", 57),
+            round2!("$t2", "$t3", "$t4", "$t0", "$t1", 58),
+            round2!("$t1", "$t2", "$t3", "$t4", "$t0", 59),
+            round3!("$t0", "$t1", "$t2", "$t3", "$t4", 60),
+            round3!("$t4", "$t0", "$t1", "$t2", "$t3", 61),
+            round3!("$t3", "$t4", "$t0", "$t1", "$t2", 62),
+            round3!("$t2", "$t3", "$t4", "$t0", "$t1", 63),
+            round3!("$t1", "$t2", "$t3", "$t4", "$t0", 64),
+            round3!("$t0", "$t1", "$t2", "$t3", "$t4", 65),
+            round3!("$t4", "$t0", "$t1", "$t2", "$t3", 66),
+            round3!("$t3", "$t4", "$t0", "$t1", "$t2", 67),
+            round3!("$t2", "$t3", "$t4", "$t0", "$t1", 68),
+            round3!("$t1", "$t2", "$t3", "$t4", "$t0", 69),
+            round3!("$t0", "$t1", "$t2", "$t3", "$t4", 70),
+            round3!("$t4", "$t0", "$t1", "$t2", "$t3", 71),
+            round3!("$t3", "$t4", "$t0", "$t1", "$t2", 72),
+            round3!("$t2", "$t3", "$t4", "$t0", "$t1", 73),
+            round3!("$t1", "$t2", "$t3", "$t4", "$t0", 74),
+            round3!("$t0", "$t1", "$t2", "$t3", "$t4", 75),
+            round3!("$t4", "$t0", "$t1", "$t2", "$t3", 76),
+            round3!("$t3", "$t4", "$t0", "$t1", "$t2", 77),
+            round3!("$t2", "$t3", "$t4", "$t0", "$t1", 78),
+            round3!("$t1", "$t2", "$t3", "$t4", "$t0", 79),
+
+            // Update state registers
+            "ld.w    $t5, $a0, 0",  // a
+            "ld.w    $t6, $a0, 4",  // b
+            "ld.w    $t7, $a0, 8",  // c
+            "ld.w    $t8, $a0, 12", // d
+            "add.w   $t0, $t0, $t5",
+            "ld.w    $t5, $a0, 16", // e
+            "add.w   $t1, $t1, $t6",
+            "add.w   $t2, $t2, $t7",
+            "add.w   $t3, $t3, $t8",
+            "add.w   $t4, $t4, $t5",
+
+            // Looping over blocks
+            "addi.d  $a1, $a1, 64",
+            "addi.d  $a2, $a2, -1",
+            "bnez    $a2, 42b",
+
+            // Save updated state
+            "st.w    $t0, $a0, 0",
+            "st.w    $t1, $a0, 4",
+            "st.w    $t2, $a0, 8",
+            "st.w    $t3, $a0, 12",
+            "st.w    $t4, $a0, 16",
+
+            // Restore stack register
+            "addi.d  $sp, $sp, 64",
+
+            in("$a0") state,
+            inout("$a1") blocks_ptr,
+            inout("$a2") blocks_len,
```

Comment:
IIUC it would be incorrect to use:
```rust
in("$a1") blocks.as_ptr(),
in("$a2") blocks.len(),
```

According to the [reference](https://doc.rust-lang.org/reference/inline-assembly.html):

>The allocated register must contain the same value at the end of the asm code (except if a lateout is allocated to the same register).

And the register values get changed in the assembly code. Unfortunately, using `inout` means that we have to silence the unused warning later.

Reply:
Fixed by using `inout(<reg>) <in expr> => _`.